includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -853,28 +874,32 @@ protected function get_wp_query_args( $query_vars ) {
 	 * @return array|object
 	 */
 	public function query( $query_vars ) {
-		$args = $this->get_wp_query_args( $query_vars );
-
-		if ( ! empty( $args['errors'] ) ) {
-			$query = (object) array(
-				'posts'         => array(),
-				'found_posts'   => 0,
-				'max_num_pages' => 0,
-			);
-		} else {
-			$query = new WP_Query( $args );
-		}
+		$orders = array();
+		try {
+			$args = $this->get_wp_query_args( $query_vars );
+
+			if ( ! empty( $args['errors'] ) ) {
+				$query = (object) array(
+					'posts'         => array(),
+					'found_posts'   => 0,
+					'max_num_pages' => 0,
+				);
+			} else {
+				$query = new WP_Query( $args );
+			}
 
-		$orders = ( isset( $query_vars['return'] ) && 'ids' === $query_vars['return'] ) ? $query->posts : array_filter( array_map( 'wc_get_order', $query->posts ) );
+			$orders = ( isset( $query_vars['return'] ) && 'ids' === $query_vars['return'] ) ? $query->posts : array_filter( array_map( 'wc_get_order', $query->posts ) );
 
-		if ( isset( $query_vars['paginate'] ) && $query_vars['paginate'] ) {
-			return (object) array(
-				'orders'        => $orders,
-				'total'         => $query->found_posts,
-				'max_num_pages' => $query->max_num_pages,
-			);
+			if ( isset( $query_vars['paginate'] ) && $query_vars['paginate'] ) {
+				return (object) array(
+					'orders'        => $orders,
+					'total'         => $query->found_posts,
+					'max_num_pages' => $query->max_num_pages,
+				);
+			}
+		} catch ( \Exception $exception ) {
+			wc_caught_exception( $exception, __METHOD__ );
```

Comment:
this function is deprecated (or at least is part of the file where functions go to die). let's throw the error.

Reply:
Oh hmm, then ya it shouldn't be used. I'll leave it up to you folks how you implement error handling here.